contracts/ERC721A.sol

```diff
@@ -62,19 +72,42 @@ contract ERC721A is Context, ERC165, IERC721, IERC721Metadata, IERC721Enumerable
         _symbol = symbol_;
     }
 
+    /**
+     * @dev Skips the zero index.
+     * This method must be called before any mints (e.g. in the constructor).
+     */
+    function _initOneIndexed() internal {
+        require(currentIndex == 0, "ERC721A: 0 index already occupied.");
+        currentIndex = 1;
+        totalBurned = 1;
+        _ownerships[0].burned = true;
+    }
+
     /**
      * @dev See {IERC721Enumerable-totalSupply}.
      */
     function totalSupply() public view override returns (uint256) {
-        return currentIndex;
+        return currentIndex - totalBurned;
```

Comment:
totalSupply is often used in write functions, might be better to just track a new storage var `totalSupply` instead of performing two SLOAD here

Reply:
probably pack the two together?

Reply:
Looks like it can help reduce the gas for some functions, especially `_burn`.

For those functions that have increased gas usage, the increase is minimal (< 50 gas).

**Before:**
https://github.com/Vectorized/ERC721A/blob/33d4b4115b3018e8689f19adfdb03c582cd6d368/contracts/ERC721A.sol#L187
```
·-----------------------------------------------------------|---------------------------|-------------|
|                   Solc version: 0.8.11                    ·  Optimizer enabled: true  ·  Runs: 800  ·
····························································|···························|·············|
|  Methods                                                  ·              100 gwei/gas               ·
······································|·····················|·············|·············|·············|
|  Contract                           ·  Method             ·  Min        ·  Max        ·  Avg        ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableMock                ·  burn               ·      49666  ·     118934  ·      97307  ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableMock                ·  initOneIndexed     ·          -  ·          -  ·      87709  ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableMock                ·  safeMint           ·      98764  ·     115864  ·     106811  ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableOwnersExplicitMock  ·  burn               ·      49666  ·      90310  ·      74260  ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableOwnersExplicitMock  ·  initOneIndexed     ·          -  ·          -  ·      87709  ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableOwnersExplicitMock  ·  safeMint           ·      77511  ·      94611  ·      82716  ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableOwnersExplicitMock  ·  setOwnersExplicit  ·      82283  ·      82480  ·      82382  ·
······································|·····················|·············|·············|·············|
|  ERC721AMock                        ·  approve            ·          -  ·          -  ·      51137  ·
······································|·····················|·············|·············|·············|
|  ERC721AMock                        ·  mint               ·      92806  ·     101402  ·      99110  ·
······································|·····················|·············|·············|·············|
|  ERC721AMock                        ·  safeMint           ·      79870  ·     125334  ·      89917  ·
······································|·····················|·············|·············|·············|
|  ERC721AMock                        ·  safeTransferFrom   ·          -  ·          -  ·      94319  ·
······································|·····················|·············|·············|·············|
|  ERC721AMock                        ·  setApprovalForAll  ·          -  ·          -  ·      46219  ·
······································|·····················|·············|·············|·············|
|  ERC721AMock                        ·  transferFrom       ·          -  ·          -  ·      87218  ·
······································|·····················|·············|·············|·············|
|  ERC721AOwnersExplicitMock          ·  safeMint           ·      79858  ·      94611  ·      85566  ·
······································|·····················|·············|·············|·············|
|  ERC721AOwnersExplicitMock          ·  setOwnersExplicit  ·      48787  ·     126038  ·      86045  ·
······································|·····················|·············|·············|·············|
```

After:

```
·-----------------------------------------------------------|---------------------------|-------------|
|                   Solc version: 0.8.11                    ·  Optimizer enabled: true  ·  Runs: 800  ·
····························································|···························|·············|
|  Methods                                                  ·              100 gwei/gas               ·
······································|·····················|·············|·············|·············|
|  Contract                           ·  Method             ·  Min        ·  Max        ·  Avg        ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableMock                ·  burn               ·      47575  ·     100017  ·      90530  ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableMock                ·  initOneIndexed     ·          -  ·          -  ·      65621  ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableMock                ·  safeMint           ·      98953  ·     116053  ·     107000  ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableOwnersExplicitMock  ·  burn               ·      47563  ·      88269  ·      67919  ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableOwnersExplicitMock  ·  initOneIndexed     ·          -  ·          -  ·      65621  ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableOwnersExplicitMock  ·  safeMint           ·      77700  ·      94800  ·      82905  ·
······································|·····················|·············|·············|·············|
|  ERC721ABurnableOwnersExplicitMock  ·  setOwnersExplicit  ·      82219  ·      82416  ·      82318  ·
······································|·····················|·············|·············|·············|
|  ERC721AMock                        ·  approve            ·          -  ·          -  ·      50942  ·
······································|·····················|·············|·············|·············|
|  ERC721AMock                        ·  mint               ·      92995  ·     101591  ·      99299  ·
······································|·····················|·············|·············|·············|
|  ERC721AMock                        ·  safeMint           ·      80059  ·     125523  ·      90106  ·
······································|·····················|·············|·············|·············|
|  ERC721AMock                        ·  safeTransferFrom   ·          -  ·          -  ·      94142  ·
······································|·····················|·············|·············|·············|
|  ERC721AMock                        ·  setApprovalForAll  ·          -  ·          -  ·      46219  ·
······································|·····················|·············|·············|·············|
|  ERC721AMock                        ·  transferFrom       ·          -  ·          -  ·      87041  ·
······································|·····················|·············|·············|·············|
|  ERC721AOwnersExplicitMock          ·  safeMint           ·      80047  ·      94800  ·      85755  ·
······································|·····················|·············|·············|·············|
|  ERC721AOwnersExplicitMock          ·  setOwnersExplicit  ·      48902  ·     125692  ·      85879  ·
······································|·····················|·············|·············|·············|
```